src/consumer/mod.rs

```diff
@@ -345,11 +343,10 @@ where
     }
 }
 
-impl<S, E, F> Consumer<S, F>
+impl<S, E> Consumer<S, E>
 where
     S: Read + Write + Reconnect + Send + 'static,
-    E: StdError,
-    F: Fn(Job) -> Result<(), E> + Send + Sync + 'static,
+    E: StdError + 'static,
```

Comment:
Huh, curious that `E` needs to be `'static` here now. I don't think it's an issue, just curious as to why.

Reply:
Good question!  I don't remember, I'll check.

Reply:
If it's not `'static` I get the following:

```
   --> src/consumer/mod.rs:422:20
    |
346 | impl<S, E> Consumer<S, E>
    |         - help: consider adding an explicit lifetime bound `E: 'static`...
...
422 |                 Ok(thread::spawn(move || {
    |                    ^^^^^^^^^^^^^
    |
note: ...so that the type `[closure@src/consumer/mod.rs:422:34: 431:18 status:std::sync::Arc<std::sync::atomic::AtomicUsize>, w:consumer::Consumer<S, E>, worker:usize, queues:std::vec::Vec<std::string::String>]` will meet its required lifetime bounds
   --> src/consumer/mod.rs:422:20
    |
422 |                 Ok(thread::spawn(move || {
    |                    ^^^^^^^^^^^^^

```

However I'm not quite sure how/whether to restructure it to fix that...

Reply:
Ah, interesting -- so it's because the error may need to be propagated back across the thread boundary. `'static` seems fine then :)